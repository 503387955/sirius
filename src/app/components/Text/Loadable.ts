import { lazyLoad } from 'utils/loadable';

const Text = lazyLoad(
  () => import('./index'),
  module => module.default,
);

export default Text;
