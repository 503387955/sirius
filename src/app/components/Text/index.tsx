import React from 'react';
import PropTypes from 'prop-types';
import { Tooltip, Text } from '@cfxjs/react-ui';
import { TooltipProps } from '@cfxjs/react-ui/dist/tooltip/tooltip';
import { TextProps } from '@cfxjs/react-ui/dist/text/text';
import styled from 'styled-components/macro';

type MyTextProps = {
  children?: React.ReactNode;
  maxWidth?: string;
  maxCount?: number;
  hoverValue?: React.ReactNode;
  tooltipConfig?: Partial<TooltipProps>;
} & Partial<TextProps>;
type NativeAttrs = Omit<React.HTMLAttributes<any>, keyof MyTextProps>;
export declare type Props = MyTextProps & NativeAttrs;

const Wrapper = styled(Text)<{ maxWidth: string }>`
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  max-width: ${props => props.maxWidth};
  display: ${props =>
    props.maxWidth === undefined ? 'inherit' : 'inline-block'};
  cursor: pointer;
` as typeof Text & React.FC<Props & { maxWidth?: string }>;

// note:
// 1. maxWidth priority is higher than maxCount
// 2. maxCount obly apply to string
// 3. if hoverValue if provide, use hoverValue as Tooltip text, other wise use children
const TextComponent = ({
  children,
  maxWidth,
  maxCount,
  hoverValue,
  tooltipConfig,
  ...props
}: Props) => {
  let child: any = children || '';
  if (maxWidth === undefined && maxCount && typeof children === 'string') {
    child = child.substr(0, maxCount) + '...';
  }
  const tooltipValue = hoverValue || children;
  return (
    <Tooltip {...tooltipConfig} text={tooltipValue}>
      <Wrapper maxWidth={maxWidth} {...props}>
        {child}
      </Wrapper>
    </Tooltip>
  );
};

TextComponent.defaultProps = {
  maxCount: 8,
};

TextComponent.propTypes = {
  maxCount: PropTypes.number,
  maxWidth: PropTypes.string,
  hoverValue: PropTypes.node,
  tooltipConfig: PropTypes.object,
};

export default TextComponent;
