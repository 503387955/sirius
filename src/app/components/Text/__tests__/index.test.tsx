import React from 'react';
import { render } from '@testing-library/react';
import Text from '..';

describe('<Tabs  />', () => {
  it('should match snapshot', () => {
    const loadingIndicator = render(
      <Text maxWidth={'40px'}>
        <span>1234567890</span>
      </Text>,
    );
    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });
});
