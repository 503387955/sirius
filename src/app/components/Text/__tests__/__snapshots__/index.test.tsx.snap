// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs  /> should match snapshot 1`] = `
.c0 {
  color: red;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  max-width: 40px;
  display: inline-block;
  cursor: pointer;
}

<div
  class="jsx-418573366 tooltip "
>
  <span
    class="c0"
  >
    <span>
      1234567890
    </span>
  </span>
</div>
`;

exports[`<Tabs  /> should match snapshot when set maxCount 1`] = `
.c0 {
  color: red;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  display: inherit;
  cursor: pointer;
}

<div
  class="jsx-418573366 tooltip "
>
  <span
    class="c0"
  >
    12345...
  </span>
</div>
`;
